public/views/VariableGrouping.vue

```diff
@@ -503,31 +515,70 @@ export default Vue.extend({
       // generate the grouping structure that describes how the vars will be grouped
       const idCol = this.isTimeseries ? getComposedVariableKey(ids) : null;
 
-      const hiddenCols = gotoTarget ? this.getHiddenCols(idCol) : [];
-
+      const hidden = gotoTarget
+        ? this.getHiddenCols(idCol, this.xCol, this.yCol)
+        : [];
       const grouping: Grouping = {
         type: this.groupingType,
         dataset: this.dataset,
         idCol: idCol,
         subIds: ids,
-        hidden: hiddenCols,
+        hidden: hidden,
       };
 
+      const groupings = [] as Grouping[];
+
       if (this.isTimeseries) {
-        const tsGrouping = grouping as TimeseriesGrouping;
+        const tsGrouping = _.cloneDeep(grouping) as TimeseriesGrouping;
```

Comment:
could we get away with just a [ ...grouping ] spread here or was it not deep enough?